src/jit/emitxarch.cpp

```diff
@@ -2918,9 +2918,25 @@ regNumber emitter::emitInsBinary(instruction ins, emitAttr attr, GenTree* dst, G
         lclVar = dst->AsLclVar();
     }
 
+    // find contained spill tmp if any
+    TempDsc* tmpDsc = nullptr;
+    if (src->isContainedSpillTemp())
+    {
+        assert(src->IsRegOptional());
+        tmpDsc = codeGen->getSpillTempDsc(src);
+    }
+    else if (dst->isContainedSpillTemp())
+    {
+        assert(dst->IsRegOptional());
+        tmpDsc = codeGen->getSpillTempDsc(dst);
+    }
+
```

Comment:
As we discussed, it is unfortunate that so much of this has spilled over into the `emitter`, as it is really more `CodeGen`-like. I'm not sure this rises to the level of an issue, but you might consider opening an issue to get crisper about whether this kind of functionality belongs in `emitter` or `CodeGen`